pkg/services/rpcsrv/server_test.go

```diff
@@ -3331,3 +3331,28 @@ func BenchmarkHandleIn(b *testing.B) {
 {"type": "Integer", "value": "42"}, {"type": "Boolean", "value": false}]]}`))
 	})
 }
+
+func TestFailedPreconditionShutdown(t *testing.T) {
+	_, srv, _ := initClearServerWithCustomConfig(t, func(c *config.Config) {
+		c.ApplicationConfiguration.RPC.Addresses = []string{"not an address"}
+	})
+
+	var stopped bool
+
+	srv.onShutdown = func() {
+		stopped = true
+	}
+
+	srv.Start()
+	require.Positive(t, len(srv.errChan)) // this is how Start reports internal failures
+	require.False(t, stopped)
+
+	go srv.Shutdown()
```

Comment:
What if you just run this synchronously?

Reply:
https://pkg.go.dev/github.com/stretchr/testify/require#Eventually + maybe some wrapper as mentioned above by @AnnaShaleva.

Reply:
done